src/horse.go

```diff
@@ -35,14 +35,10 @@ func (h *Horse) Clone(other *Horse) {
 	h.Line = other.Line
 }
 
-func (h Horse) Letter() string {
+func (h *Horse) Letter() string {
 	return fmt.Sprintf("%c", h.Name[0])
 }
```

Comment:
I thought (as far as I have learned so far) it's better to not use the pointer when it's not necessary. And in the case for functions that are not altering the struct itself, so you just want to query its data, then the value would be better.  

I found this in the [official doc](https://go.dev/doc/effective_go#printing) [search for `String()`]
```go
type MyString string
func (m MyString) String() string {
    return fmt.Sprintf("MyString=%s", string(m)) // OK: note conversion.
}
```
What do you think?

Reply:
Yes, if you don't plan to mutate an object then you don't need pointer receivers. 

The reason I changed it to * was to be consistent and have all methods same receiver type. My IDE was complaining about that, but I don't know if there is a standard here and if it's OK/no to have different receiver for same type. This is worth investigating.

Personally, if I would get rid of that Clone() method somehow, then you have only value receivers

Reply:
Yeah, that `Clone()` smell also fishy to me, it will worth keep investigating, and see if we figure out a way to get rid of it. Anyway, I don't have personally any preference -also I dont really know what's best or not- so I am OK with your suggestion, as I dont have strong arguments but those that I already presented to you.